server/src/main/java/org/elasticsearch/action/admin/cluster/state/ClusterStateAction.java

```diff
@@ -9,13 +9,19 @@
 package org.elasticsearch.action.admin.cluster.state;
 
 import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.RemoteClusterActionType;
+import org.elasticsearch.common.io.stream.Writeable;
 
 public class ClusterStateAction extends ActionType<ClusterStateResponse> {
 
     public static final ClusterStateAction INSTANCE = new ClusterStateAction();
     public static final String NAME = "cluster:monitor/state";
+    public static final RemoteClusterActionType<ClusterStateResponse> REMOTE_TYPE = new RemoteClusterActionType<>(
+        NAME,
+        ClusterStateResponse::new
+    );
 
     private ClusterStateAction() {
-        super(NAME, ClusterStateResponse::new);
+        super(NAME, Writeable.Reader.localOnly());
```

Comment:
Nit: I prefer the style in [PainlessExecuteAction.java](https://github.com/elastic/elasticsearch/pull/104611/files#diff-ff9ff526027147221fc0f045907c7df06ca43510d6f77ee52350417a0fae8982); I think it's a bit confusing to have ClusterStateAction be a "local" `ActionType` (constructed with `localOnly`) and have a `RemoteClusterActionType` field.

Reply:
Yeah again there's lots of legacy in this area, I'd prefer not to subclass `ActionType` too (see #104139) but cleaning that up is a whole other marathon task.

Reply:
Yes, makes sense to do it separately (if ever)